src/main/scala/com/github/mrpowers/spark/fast/tests/SchemaComparer.scala

```diff
@@ -4,13 +4,20 @@ import org.apache.spark.sql.types.{StructField, StructType}
 
 object SchemaComparer {
 
-  def equals(s1: StructType, s2: StructType, ignoreNullable: Boolean = false): Boolean = {
+  def equals(
+    s1: StructType,
+    s2: StructType,
+    ignoreNullable: Boolean = false,
+    ignoreColumnNames: Boolean = false
+  ): Boolean = {
     if (s1.length != s2.length) {
       false
     } else {
       val structFields: Seq[(StructField, StructField)] = s1.zip(s2)
       structFields.forall { t =>
-        t._1.name == t._2.name && t._1.dataType == t._2.dataType
+        ((t._1.nullable == t._2.nullable) || ignoreNullable) &&
+        ((t._1.name == t._2.name) || ignoreColumnNames) &&
```

Comment:
We could extract out expressions and assign meaningful names, so that newbies won't have a hard time understanding it. Just a thought.

Reply:
I will leave it up to you :) Doesn't seem necessary to me but I'm obviously biased.

Reply:
Lets leave it like this for now.